docs/arguments.md

```diff
@@ -2,20 +2,18 @@
 
 ## The `Arguments` class
 
-The `Arguments` class is used by Windsor to pass arguments [down the invocation pipeline](how-dependencies-are-resolved.md). The class is a simple implementation of non-generic `IDictionary`, but it has some useful capabilities.
-
-:information_source: **Custom `IDictionary` is respected:** When you call `container.Resolve` passing your own custom implementation of `IDictionary` Windsor will respect that and not replace it with `Arguments`. It is advised that you use `Arguments` though.
+The `Arguments` class is used by Windsor to pass arguments [down the invocation pipeline](how-dependencies-are-resolved.md). The class is a simple implementation of non-generic `IDictionary`, but it has some useful capabilities explained further below.
 
 ### Constructors
 
 The class has several constructors:
 
 #### `namedArgumentsAsAnonymousType`
```

Comment:
@jonorossi 

This documentation is absolute shyte. 

I think what we need here, is a concise clear document that preludes with a definition of what typed/named arguments are. We follow with a body of methods usages/examples.

Reply:
> I think what we need here, is a concise clear document that preludes with a definition of what typed/named arguments are in terms of dependency resolution(including examples). Then we follow with a body of instance methods including usages/examples.

As usual documentation can always be improved. I'll let you decide when to do that, before or after merge.

I'll do another code review to make sure we are good to merge.

Reply:
Nope. Not right. Let's release ASP.NET Core first. I want to fix this properly.